### ISSUE TITLE
Camera feed stops showing in HA

### ISSUE THREAD

**Issue by xHackYx** [fwestenberg/reolink_dev#181]
Hey there,

sorry for my bad english at first :) 

3 of my 5 Cams stop showing in lovelace about a half day after i restart my ha.

Im using 5x B800 with RLN8-410-E NVR.

Looks like this:

![image](https://user-images.githubusercontent.com/67443272/113114982-3a481f00-920c-11eb-8fb1-c12bf3cd9312.png)

If i restart HA its working for half a day +-
This Problem is new to me but the reolink integration never worked perfect for me. Normally the livefeed of the of the cams stopped working and it was just showing a picture.

I already reinstalled the integration. NVR Software and HA are Uptodate. Cams working fine with reolink app.
The weired think is that not the same cameras are not showing. its just randomly sometimes camera 1,2 working sometimes 4,5 and so on.

`Logger: custom_components.reolink_dev.base
Source: custom_components/reolink_dev/base.py:267
Integration: Reolink IP camera (documentation, issues)
First occurred: March 30, 2021, 1:16:50 PM (2544 occurrences)
Last logged: 10:21:54 AM

Host 10.0.0.155 error renewing the Reolink subscription`

`Logger: homeassistant.components.stream.worker
Source: components/stream/worker.py:130
Integration: Stream (documentation, issues)
First occurred: 8:42:58 AM (21 occurrences)
Last logged: 10:19:06 AM

Error opening stream rtmp://10.0.0.155:1935/bcs/channel3_sub.bcs?channel=3&stream=0&token=b5fda0b809795ab
Error opening stream rtmp://10.0.0.155:1935/bcs/channel7_sub.bcs?channel=7&stream=0&token=f43818dceba9bd4
Error opening stream rtmp://10.0.0.155:1935/bcs/channel5_sub.bcs?channel=5&stream=0&token=8f924ea23b46a94`

**Comment by miked314** [comment]
I'm having similar issues with RLC-822A and RLC-520A.
RLC-822A (IP ends with 86)  is 4k so i have it setup to play sub channel. 
it works after reloading HA core for some time. after that i see this:

Error opening stream rtmp://192.168.1.86:1935/bcs/channel0_sub.bcs?channel=0&stream=0&token=ef35fe90f05e95e
Error opening stream rtmp://192.168.1.20:1935/bcs/channel0_main.bcs?channel=0&stream=0&token=59902a33407806b

if I open camera webpage and authenticate and grab TOKEN from POST I'm able to view stream in VLC with rtmp command listed above. 
Im also able to stream in VLC without TOKEN but with user=USERNAME&password=PASSWORD.

it seems like there is an issue with authenticating maybe? 

![image](https://user-images.githubusercontent.com/27921824/113234367-bcfdc800-9266-11eb-8c2d-702b4725ad83.png)

![image](https://user-images.githubusercontent.com/27921824/113234414-d69f0f80-9266-11eb-81d2-a5ec3147557c.png)

**Comment by Pigotka** [comment]
Same issue here. But sometimes it renews itself. But after I added RLC-822A the live view is always in some error.

**Comment by toreamun** [comment]
I have a similar issue and have to reload the integration

Error opening stream rtmp://172.16.2.10:1935/bcs/channel0_main.bcs?channel=0&stream=0&token=bc02207f8ae22af

**Comment by tdaime** [comment]
Same issue with a RLC510A
Works after reload an after some time stops working, with the following error: 
Error opening stream rtmp://192.168.1.55:1935/bcs/channel0_sub.bcs?channel=0&stream=0&token=14cbb2402970191

Tried different combinations (RTMP/RTSP ,  Main/Sub) but still same behavior

**Comment by andriej** [comment]
Similar with two 520 too.

**Comment by jchasey** [comment]
Seeing the same issue

**Comment by xHackYx** [comment]
Do guy also have the problem, that your live view gets stuck ? Some of my cams just freezing

**Comment by andriej** [comment]
Yes. What bugs me even more is that it's PoE over LAN, so theoretically should work flawlessly...

**Comment by danbayliss** [comment]
I'm seeing the same. I've got five cameras, two show a live view, two show a repeating five second snippet from five hours ago and another shows a five second clip from seven hours ago. Only way to fix it is to reboot

**Comment by tmooz** [comment]
same with 510 and 520s.

**Comment by Trebacz** [comment]
I think i have the same issue, but just reloading the integration seems to "fix" it. 
![image](https://user-images.githubusercontent.com/19363347/122684755-3c42ce00-d1bc-11eb-8179-423ff4754ee6.png)

**Comment by Pigotka** [comment]
@Trebacz yes but only for next one or two stream views.

**Comment by rizz360** [comment]
> user=USERNAME&password=PASSWORD

It would be great having this as an option in the configuration than. I cannot rely on this integration if I have to keep reloading it 1-2 times a day...

**Comment by fwestenberg** [comment]
Please update to v0.18 and let me know. It now uses username+password instead of token. The token seems to break the stream after a while.

**Comment by tdaime** [comment]
> Please update to v0.18 and let me know. It now uses username+password instead of token. The token seems to break the stream after a while.

Thanks, working fine for me now.

**Comment by rizz360** [comment]
> Please update to v0.18 and let me know. It now uses username+password instead of token. The token seems to break the stream after a while.

Same here! Since the update the feed never stopped. Before it stopped after checking it a few times. Thank you!

**Comment by Trebacz** [comment]
> Please update to v0.18 and let me know. It now uses username+password instead of token. The token seems to break the stream after a while.

Working here also - Thank you!

**Comment by xHackYx** [comment]
sadly im still faceing the same issu with 0.18.

LOG:
```
Logger: custom_components.reolink_dev.base
Source: custom_components/reolink_dev/base.py:339 
Integration: Reolink IP camera (documentation, issues) 
First occurred: 2:16:21 AM (5 occurrences) 
Last logged: 2:16:21 AM

Host 192.168.4.100 error renewing the Reolink subscription
```

![E9419823-5E7C-4C45-815A-FB806AA8FE34_1_102_o](https://user-images.githubusercontent.com/67443272/126913146-070401d6-e3c1-4965-b514-ebf92d82a00f.jpeg)

As u can see one feed is working. the one in night mode. Its currently 22:40
feeds stuck at 2x 20:53 1x 18:25
